ctparse/nb_estimator.py

```diff
@@ -0,0 +1,74 @@
+from typing import Sequence, Dict, Tuple, List
+from math import log, exp
+
+
+def _log_sum_exp(x: Sequence[float]) -> float:
+    max_value = max(x)
+    sum_of_exp = sum(exp(x_i - max_value) for x_i in x)
+    return max_value + log(sum_of_exp)
+
+
+class MultinomialNaiveBayes:
+    def __init__(self, alpha: float = 1.0):
+        self.alpha = alpha
+        self.class_prior = (0.0, 0.0)
+        self.log_likelihood: Dict[str, List[float]] = {}
+
+    def _construct_log_class_prior(self, y: Sequence[int]) -> None:
+        # Input classes are -1 and 1
+        neg_class_count = sum(1 if y_i == -1 else 0 for y_i in y)
+        pos_class_count = len(y) - neg_class_count
+
+        neg_log_prior = log(neg_class_count / (pos_class_count + neg_class_count))
+        pos_log_prior = log(pos_class_count / (pos_class_count + neg_class_count))
+        self.class_prior = (neg_log_prior, pos_log_prior)
```

Comment:
why setting instead of returning? It could be a sklearn style thing, I think it would be better to return and set outside (single responsibility principle). Matter of style, not critical